### PR TITLE
Fix #7331: handle permission error in search for project file

### LINKED ISSUE #7331: Search for project root crashes when (parent) directory lacks permissions
Tested on Nix-on-Droid with Agda 2.6.4 and Debian trixie/sid with Agda 2.6.3-1, Debian trixie/sid with cabal built Agda 2.6.4.3. 

Sample setup (to emulate problem on a typical Linux system):

    sudo chmod o-r /     # Remember later to revert with sudo chmod +r /
    mkdir -p testdir/hello
    chmod -r testdir
    cd testdir/hello
    echo 'main = putStrLn "hello"' > hello.agda

Tried to run agda hello.agda. Immediately errors with:

    /some/path/above: getDirectoryContents:openDirStream: permission
    denied (Permission denied)

Note the complete absence of indication what Agda is looking for. This isn't indicated in the installation or hello world instructions. Searched around and found documentation hinting at a "project root" and .agda-lib files, so tried creating a hello.agda-lib:

    echo >hello.agda-lib name: hello

Result:

    /: getDirectoryContents:openDirStream: permission denied
    (Permission denied)

I can only guess this isn't the project root search because it's targeting the directory differently. The compiler just expects global system access without motivation. There's no indication where in the compiler this is or why it's not handled.

### PR DESCRIPTION
**Pull request by andreasabel**
Closes #7331.

### CODE REVIEW

File: test/interaction/Issue7331.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+mkdir -p Issue7331/dir              > /dev/null
+chmod +r Issue7331                  > /dev/null
+touch Issue7331/dir/Issue7331.agda  > /dev/null
+cd Issue7331/dir                    > /dev/null
+
+# Drop permissions for parent dir
+chmod -r ..                         > /dev/null
+
+${AGDA} Issue7331.agda
```
Comment (UlfNorell) [inline test/interaction/Issue7331.sh]: Should the test clean up the directory?

File: test/interaction/Issue7331.agda
Comment (UlfNorell) [inline test/interaction/Issue7331.agda:1]: Accidental commit, or does it need to be there for technical reasons?

**Review by UlfNorell — APPROVED** [review]

File: test/interaction/Issue7331.agda
Comment (andreasabel) [reply test/interaction/Issue7331.agda:1]: I needed it for `make Issue7331.cmp`, but maybe I was mistaken that it is needed in general...
I can delete it.

Comment (andreasabel) [reply test/interaction/Issue7331.agda:1]: I think it is needed: https://github.com/agda/agda/blob/37718bf47402201b56475ed4d4ae961d05a6b090/test/interaction/Makefile#L110-L111

File: test/interaction/Issue7331.sh (hunk 1 above)
Comment (andreasabel) [reply test/interaction/Issue7331.sh]: I amended the commit to add cleaning.

**Review by UlfNorell — APPROVED** [review]

**Pull request merged**